File: 62-unique-paths/62-unique-paths.java
class Solution {
    public int uniquePaths(int m, int n) {  
        int[][] dp = new int[m+1][n+1];
        dp[1][1] = 1;
        
        return dfs(dp, m, n);   
    }
    
    private int dfs(int[][] dp, int row, int col){
        if(row == 0 || col == 0){
            return 0;
        }
        if(dp[row][col] != 0){
            return dp[row][col];
        }else{
            int waysThere =  dfs(dp,row-1, col) + dfs(dp, row, col-1);
            dp[row][col] = waysThere;
            return dp[row][col];
        }
    }
}

File: 62-unique-paths/62-unique-paths.py
class Solution:
    def uniquePaths(self, m: int, n: int) -> int:
        dp = [[0] * n for i in range(m)]
        for row in range(m):
            for col in range(n):
                if row == 0 or col == 0:
                    dp[row][col] = 1
                else:
                    up = dp[row-1][col]
                    left = dp[row][col-1]
                    dp[row][col] = up + left
                    
        return dp[m-1][n-1]